crates/bevy_ecs/macros/src/lib.rs

```diff
@@ -254,6 +254,30 @@ pub fn impl_param_set(_input: TokenStream) -> TokenStream {
                 fn apply(&mut self, world: &mut World) {
                     self.0.apply(world)
                 }
+
+                fn world_access_level() -> WorldAccessLevel {
+                    let mut exclusive = false;
+                    let mut shared = false;
+                    #(
+                        match #param_fetch::world_access_level() {
+                            WorldAccessLevel::Exclusive => {
+                                exclusive = true;
```

Comment:
Doesn't this need to panic if there's already shared access?

Reply:
Nope, this is `ParamSet`, different params are allowed to conflict.